packages/server/src/restaurants/database/DishesSchema.ts

```diff
@@ -0,0 +1,185 @@
+import { Prop, raw, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Exclude, Expose } from 'class-transformer';
+import { ObjectId } from 'mongodb';
+import mongoose, { Document } from 'mongoose';
+
+type DishDocument = Dish & Document<ObjectId>;
+
+enum DishTags {
+  Vegan = 'wegańska',
+  Vegetarian = 'wegetariańska',
+  GlutenFree = 'gluten free',
+  Spicy = 'ostre',
+  VerySpicy = 'bardzo ostre',
+}
+
+enum MealType {
+  Breakfast = 'śniadanie',
+  Lunch = 'lunch',
+  Dinner = 'obiad',
+  Supper = 'kolacja',
+}
+
+enum Allergens {
+  Gluten = 'gluten',
+  ShellFish = 'skorupiaki',
+  Eggs = 'jaja',
+  Fish = 'ryby',
+  Nuts = 'orzechy',
+  Soya = 'soja',
+  Milk = 'mleko',
+  Celery = 'seler',
+  Mustard = 'gorczyca',
+  SesameSeeds = 'nasiona sezamu',
+  SulphurDioxide = 'dwutlenek siarki',
+  Lupine = 'łubin',
+  Molluscs = 'mięczaki',
+}
+
+const DISH_CONSTANTS = Object.freeze({
+  DISH_LEN_STR: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 500,
+  }),
+  DISH_LEN_NUMBER: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 5,
```

Comment:
Z tego co rozumiem to walidatory `min` i `max` dla Number określają min/max wartość, a nie liczbę cyfr, więc 5 to chyba trochę mało 😀

Reply:
Prawda, w ogóle chyba to ograniczenie nie jest potrzebne, bo ono ma zabezpieczać nas przed zaspamowaniem bazy danych dużą ilością danych, a liczba `1` i `100000` zajmuje tyle samo bajtów pamięci więc to nie jest aż taki problem jak ze stringami. Min 0, żeby nie pozwolić na ujemne wartości chyba wystarczy.